lib/salus/scanners/semgrep.rb

```diff
@@ -209,7 +207,7 @@ def build_command_and_message(match, strict, base_path, exclude_directories_flag
     def message_from_hit(hit, match)
       has_external_config = !match['config'].nil?
       msg = if has_external_config
-              hit['extra']['message']
+              hit['extra']['message'] + "\n\trule_id: " + hit['check_id']
```

Comment:
I'm trying to understand this change. Can you walk me through it? 🤔

Reply:
In the new semgrep version, there is a whitelisting feature that allows you to whitelist rule ids.
But in the previous salus version, if a forbidden pattern is found, it doesn't tell you what the rule id is, like

```
  Forbidden patterns in config "semgrep_rules.yaml" was found 
  	test.js:      password = "hardcoded_secret"```
```

So I changed the output to include the rule_id, like
```
  Forbidden patterns in config "semgrep_rules.yaml" was found 
        rule_id: hardcoded_secret
  	test.js:      password = "secret"
```

This way, the user can whitelist the rule id like


```
password = "secret" // nosem: hardcoded_secret
```

Reply:
Does the `check_id` need to be in the message text? It seems like it would be more useful as a separate field. Also, it's not clear why you are checking for `has_external_config` to determine if the message should be changed?

Reply:
@cbtoni, did you mean make the output look like below?
{code}
  Forbidden patterns in config "semgrep_rules.yaml" was found 
  	test.js:10:      password = "secret"
  Forbidden patterns in config "semgrep_rules.yaml" was found 
  	test.js:20:     if (0 == 0) ...

  Forbidden pattern rule_ids: hardcoded_secret, useless comparison
{code}

`has_external_config` is used when you reference semgrep config files in salus.yaml.  Too add a semgrep rule, you can either reference a semgrep config file in salus.yaml, or add the rule directly. But if you add the rule directly, it only works with simple rules and the `rule_id` option is not supported.

Reply:
I think I'm a little confused about how the messages are put together in the output from reading the specs, and I thought there would be one entry for each hit, with a message and a rule_id. If that's not the case then you can ignore me.

Reply:
@cbtoni There is one entry for each hit, with a message and a rule_id, like below. How did you want `rule_id` to be a separate field in the output below?

```
==== Semgrep: FAILED in 2.87s

 ~~ Scanner Logs:


  Forbidden patterns in config "semgrep_rules.yaml" was found - Hardcoded password 1
  	rule_id: hardcoded_password_1
  	pass.js:1:password =

  Forbidden patterns in config "semgrep_rules.yaml" was found - Plesae use safeLoad if possible
  	rule_id: unsafe_yaml_load_3
  	c/t.js:4:      return <pre className="ex">{yaml.dump(yaml
  .load(definition.get("example")), defaultYamlOptions)}</pre>;

```

Reply:
@ghbren I think it's fine like that, thanks for sharing an example with me